### PR TITLE
OD-107 [Feature] Users will be able to see multiple images in the LFD detail view. 

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://weboo.atlassian.net/browse/OD-107

## Description
Added a new logic for the multiple images in the detail view.
Know users will be able to see multiple images and interact with them like in the gallery. 

## Screenshots/screencasts
https://share.getcloudapp.com/bLug6ROl

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko @ivandevupp @AndrRyaz

### CODE REVIEW

File: templates/build/small-card-detail.build.hbs [hunk 1]
```diff
@@ -54,12 +54,21 @@
         \{{#ifCond type '==' 'image'}}
           \{{#if content}}
             \{{#if labelEnabled}}
-            <div class="small-card-list-detail-label">\{{label}}</div>
+	            <div class="small-card-list-detail-label">\{{label}}</div>
```
Comment (YaroslavOvdii) [inline templates/build/small-card-detail.build.hbs:57]: image
About this indent. When I edit it in the VSCode indent is okay, and I don't know why in the git it displays wrong.

Comment (tonytlwu) [reply templates/build/small-card-detail.build.hbs:57]: @YaroslavOvdii There's probably a tab character in there somewhere.

File: js/utils.js [hunk 2]
```diff
@@ -2108,7 +2182,8 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
       isDeletable: recordIsDeletable,
       isCurrentUser: recordIsCurrentUser,
       matchesFilters: recordMatchesFilters,
-      getUniqueId: getRecordUniqueId
+      getUniqueId: getRecordUniqueId,
+      imageContent: prepareImageContent
```
Comment (tonytlwu) [inline js/utils.js:2186]:
```suggestion
      getImageContent: getImageContent
```

File: js/utils.js [hunk 3]
```diff
@@ -130,6 +130,37 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
...
+
+    imageContent = imagesArray
+      ? imagesArray[0]
+      : '';
+
+    if (isSummary) {
+      return imageContent;
+    }
+
+    return { imageContent: imageContent, imagesArray: imagesArray };
```
Comment (tonytlwu) [inline js/utils.js:161]:
```suggestion
    return {
      imageContent: imageContent,
      imageArray: imageArray
    };
```

File: js/layout-javascript/agenda-code.js [hunk 4]
```diff
@@ -2550,7 +2561,27 @@ DynamicList.prototype.addDetailViewData = function(entry) {
       content = entry.originalData[dynamicDataObj.column];
     }
 
-    content = _this.Utils.String.toFormattedString(content);
+    if (dynamicDataObj.type === 'image') {
+      var imagesContentData = _this.Utils.Record.imageContent(entry.originalData[dynamicDataObj.column]);
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js:2565]: The logic for everything in this `if` clause is still repeated. They look pretty similar, if not exactly the same in all files. This is something where if there's bug or enhancement needed, it'll become very complex to update and consider backward compatibility. It's also not something that users would really need to customise, and therefore has no reason to be repeated in the layout JS files.

File: js/layout-javascript/agenda-code.js [hunk 5]
```diff
@@ -903,6 +904,14 @@ DynamicList.prototype.attachObservers = function() {
       $parent.find('.btn-bookmarked').focus();
 
       record.bookmarkButton.like();
+    })
+    .on('click', '.multiple-images-item', function() {
```
Comment (tonytlwu) [inline js/layout-javascript/agenda-code.js:908]: Add accessibility enhancements to make sure users can open the images with keyboard.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: templates/build/small-card-detail.build.hbs (hunk 1 above)
Comment (YaroslavOvdii) [reply templates/build/small-card-detail.build.hbs:57]: @tonytlwu Yes, you were right about that. Removed tab character

**Comment by YaroslavOvdii** [comment]
@tonytlwu added suggestions

File: js/utils.js [hunk 6]
```diff
@@ -2090,6 +2178,7 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
       splitByCommas: splitByCommas,
       validateImageUrl: validateImageUrl,
       toFormattedString: toFormattedString,
+      getImagesByRegex: getImagesUrlsByRegex,
```
Comment (tonytlwu) [inline js/utils.js:2181]:
```suggestion
```

File: js/utils.js [hunk 7]
```diff
@@ -130,6 +130,52 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
     return Promise.all(formFilesInfoInDetailViewOptions);
   }
 
+  /**
+   * This function is preparing image original data to display images in the detail view
+   *
+   * @param {String | Array} content - content data that we get from originalData
+   * @param {Boolean} isSummary - flag that show us from where this function was called
+   * @returns {Object | String} in case isSummary is true than we will return only first image URL.
```
Comment (tonytlwu) [inline js/utils.js:138]:
```suggestion
   * @returns {Object | String} in case isSummary is true then we will return only the first image URL.
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/utils.js (hunk 6 above)
Comment (YaroslavOvdii) [reply js/utils.js]: @tonytlwu removed

File: js/utils.js (hunk 7 above)
Comment (YaroslavOvdii) [reply js/utils.js]: @tonytlwu fixed

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: High
Affected: All LFD layouts
Verify: LFD entries detail views are opened and rendered as expected

Squash and merge: Use PR title

**Comment by YaroslavOvdii** [comment]
@tonytlwu resolved conflicts

**Review by tonytlwu — APPROVED** [review]

File: js/utils.js [hunk 8]
```diff
@@ -1213,40 +1263,73 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
...
-          return data.record;
-        }
+            return data.record;
+          }
 
-        var urlEdited = _.some(response.files, function(file) {
+          var urlEdited = _.some(response.files, function(file) {
           // remove file extension
-          var fileName = file.name.match(/(.+?)(?:\.[^\.]*$|$)/)[1];
+            var fileName = file.name.match(fileExtensionRegex)[1];
```
Comment (tonytlwu) [inline js/utils.js:1315]: @YaroslavOvdii The `fileExtensionRegex` variable is not defined here.

Comment (YaroslavOvdii) [reply js/utils.js:1315]: @tonytlwu It's defined at this [line](https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/562/files#diff-1e20a0ee0fa5c5689cd8541a5892e5e23248b3bf2ab918043e525c443856f18cR1275)

Comment (tonytlwu) [reply js/utils.js:1315]: @YaroslavOvdii But that's in the `if {}` clause and this is the `else {}` clause, which doesn't get access to that variable. That's why it's `undefined`.

Comment (YaroslavOvdii) [reply js/utils.js:1315]: @tonytlwu Yes, you are right. It seems to be a problem. Because it was defined in the if clauser.

**Pull request merged**